src/groovy/jobs/steps/SimpleDumpTableResultStep.groovy

```diff
@@ -34,7 +36,11 @@ class SimpleDumpTableResultStep extends AbstractDumpStep {
     }
 
     void writeHeader(CSVWriter writer) {
-        writer.writeNext(headers as String[])
+        List<String> fileHeaders = headers
+        if (headerMessageClosure) {
+            fileHeaders = headers.collect { headerMessageClosure(it) }
+        }
+        writer.writeNext(fileHeaders as String[])
```

Comment:
This doesn't make sense to me. You already heave a protected `getHeaders` method. Why would you on top of that provide in the same class a method to transform those headers?